### ISSUE TITLE
Z3 hangs or gives unknown on QF_NIRA formula

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#2426]
Hi, 

For this QF_NIRA formula:
```
(declare-fun a () Int)
(declare-fun b () Int)
(declare-fun c () Int)
(declare-fun d () Int)
(assert (> (* (/ b d) 8 ) 1924 a (* (/ b d) 8 )))
(assert (= b (* c d)))
(check-sat)
```
Z3 nightly will hang on it or give unknown result, while z3-4.8.5 can report `unsat` immediately.

OS: Ubuntu 18.04
Revision: aff4b30

**Comment by NikolajBjorner** [comment]
It is my understanding that this example was synthetic and does not come from an application.
The issue is therefore deferred as a test for the new arithmetic solver and will not be addressed in the current solver

**Comment by levnach** [comment]
The latest it worked in revision: a13ac6a75 ("Merge pull request #2389 from agurfinkel/issue_2357", 2019-07-10)
In the next revision (5de35d46e ("fix #2390", 2019-07-10)) it does not work.
Suspiciously, the added code was dealing with the division of expressions.
Another thing catches the eye that in the inequality
(assert (> _(* (/ b d) 8 )_ 1924 a _(* (/ b d) 8 )_)) the first and the last expressions are the same.

**Comment by NikolajBjorner** [comment]
fixed ca11dc1fc5cee87e36e487503e49dfa198cce1b2
